opendevin/controller/agent_controller.py

```diff
@@ -230,6 +231,19 @@ async def set_agent_state_to(self, new_state: AgentState):
         ):
             # user intends to interrupt traffic control and let the task resume temporarily
             self.state.traffic_control_state = TrafficControlState.PAUSED
+            # User has chosen to deliberately continue - lets double the max iterations
+            if (
+                self.state.iteration is not None
+                and self.state.max_iterations is not None
+            ):
+                if self.state.iteration >= self.state.max_iterations:
+                    self.state.max_iterations += self._initial_max_iterations
+            if (
+                self.state.metrics.accumulated_cost is not None
+                and self.max_budget_per_task is not None
+            ):
+                if self.state.metrics.accumulated_cost >= self.max_budget_per_task:
+                    self.max_budget_per_task *= 2
```

Comment:
This has the same issue, let's say it's $2, next time it makes it $4, then $8. I think what makes most sense is the same behavior as with iterations.

Reply:
Indeed. Adapted same tweak for this now.